### PR TITLE
Add more tests for lib/node/NodeMainTemplatePlugin.js

### PR DESCRIPTION
**Pull request by jure**
**What kind of change does this PR introduce?**

It adds tests for `lib/node/NodeMainTemplatePlugin.js` as described in issue #3716.

**Did you add tests for your changes?**

Only tests.

**Summary**

This PR increases the coverage for NodeMainTemplatePlugin.js above 80%. It also adds tests for chunk filename length, in addition to the existing filename length tests. This PR was inspired by #3716.

**Does this PR introduce a breaking change?**

No.

### CODE REVIEW

**Comment by jure** [comment]
Just sweet sweet tests here, no tricky features, nothing but that good testing stuff. Merge? :)

File: test/configCases/hash-length/output-filename/test.config.js [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
-			if (bundleDetect.test(file)) {
-				file.should.match(new RegExp("^.{" + expectedNameLength + "}$"));
-				return "./" + file;
-			}
+		var bundleDetects = [{
+			regex: new RegExp("^bundle" + i, "i"),
+			expectedNameLength: options.amd.expectedFilenameLength
+		}, {
+			regex: new RegExp("^0.bundle" + i, "i"),
+			expectedNameLenght: options.amd.expectedChunkFilenameLenght
```
Comment (sokra) [inline test/configCases/hash-length/output-filename/test.config.js]: expectedNameLenght -> expectedNameLength

File: test/configCases/hash-length/output-filename/test.config.js [hunk 2]
```diff
@@ -1,17 +1,37 @@
...
+		}
```
Comment (sokra) [inline test/configCases/hash-length/output-filename/test.config.js]: formating

File: test/configCases/hash-length/output-filename/webpack.config.js [hunk 3]
```diff
@@ -44,7 +48,8 @@ module.exports = [{
 		chunkFilename: "[id].bundle4.[hash:8].js"
 	},
 	amd: {
-		expectedFilenameLength: 31
+		expectedFilenameLength: 31,
+  	expectedChunkFilenameLength: 21
```
Comment (sokra) [inline test/configCases/hash-length/output-filename/webpack.config.js]: make sure to get the spacing correct

File: test/configCases/hash-length/output-filename/test.config.js [hunk 4]
```diff
@@ -1,17 +1,37 @@
...
+		]
+
+		var bundleDetect;
+		for (bundleDetect of bundleDetects) {
+			return verifyFilenameLength(
```
Comment (sokra) [inline test/configCases/hash-length/output-filename/test.config.js]: separate the validation loop from the searching loop. Elsewise you only validate file until one is found.
Don't return on undefined.

**Comment by jure** [comment]
Thanks @TheLarkInn! My second in the webpack universe, one was already merged and released for enhanced-resolve, that one was fun (https://github.com/webpack/enhanced-resolve/pull/47)!

**Comment by TheLarkInn** [comment]
I'm going to merge this. It appears that changes have been update based on @sokra's request as well as all checks passing. codecov just not reporting for some reason.

**Comment by TheLarkInn** [comment]
Thank you so much @jure 🙇 🙇 🍾 🎉

**Pull request merged**